### ISSUE TITLE
Backtick introduced into relative source file paths in compilation database

### ISSUE THREAD

**Issue by OmegaPhil** [nickdiego/compiledb-generator#14]
Similar to [this bug](https://github.com/nickdiego/compiledb-generator/issues/6), it looks like the relative paths for source files are mangled at the start - the backtick completely breaks things here:

```
  "'./'`bar-view.cpp"
  ], 
  "file": "'./'`bar-view.cpp"
```

libtool is probably doing something weird here, as the result is a shared object ([XFCE4 panel plugin](https://git.xfce.org/panel-plugins/xfce4-hardware-monitor-plugin/) project).

Version of tool used is f0f79d5759f14b7f1e16b274e9bb2513411e0499 (doesn't look like theres an official version for these scripts).

**Comment by eklitzke** [comment]
This isn't really a libtool thing, it's something automake does. I'm trying to figure out if there's a non-gross way to fix this.

**Comment by nickdiego** [comment]
> This isn't really a libtool thing, it's something automake does. I'm trying to figure out if there's a non-gross way to fix this.

Yes, you're right. It's already been marked as a autotools-specific issue, caused when more complex shell commands are present. Unfortunately I'm currently with no time to look into this issue..
Feel free to try some solution that could solve that in a more generic way.. Maybe [something in this direction](https://github.com/nickdiego/compiledb-generator/issues/13#issuecomment-358193045) would be good enough, not sure.

Thanks in advance.

**Comment by lyonlh** [comment]
@eklitzke 
I wrote a new tool by pure shell, which intercept implicit variable CC and CXX. I think it can resolve the problem. You can give it a try.

**Comment by nickdiego** [comment]
Finally got some time to work on this.

It should be fixed as #34 has landed.
Please, @lyonlh and @eklitzke feel to test it with your codebases and give some feedback if possible. I also added a simple unit test using a real command generated by automake it seems to be working ok.

Unfortunately there is still [an issue](https://github.com/nickdiego/compiledb-generator/issues/35) when using `compiledb make` with autotools-based builds, it seems to be related to `-B / --always-make` flag, which cause `configure` script to be executed for every make target. _This issue will be addressed in a separate PR._ To work around this issue use something like:

```bash
$ make -nwk | compiledb -o-
```
Thanks.

**Comment by lyonlh** [comment]
@nickdiego Failed to install `compiledb`. Please refer to issue [#37](https://github.com/nickdiego/compiledb-generator/pull/37).
